app/src/main/kotlin/com/wire/android/ui/home/conversations/model/UIMessage.kt

```diff
@@ -367,13 +368,34 @@ sealed class UIMessageContent {
             }
         )
 
-        class HistoryLost : SystemMessage(R.drawable.ic_info, R.string.label_system_message_conversation_history_lost, true)
         class MLSWrongEpochWarning : SystemMessage(
             iconResId = R.drawable.ic_info,
             stringResId = R.string.label_system_message_conversation_mls_wrong_epoch_error_handled,
             isSmallIcon = true
         )
 
+        data class ConversationProtocolChanged(
+            val protocol: Conversation.Protocol
+        ) : SystemMessage(
+            R.drawable.ic_info,
+            when (protocol) {
+                Conversation.Protocol.PROTEUS -> R.string.label_system_message_conversation_protocol_changed_proteus
+                Conversation.Protocol.MIXED -> R.string.label_system_message_conversation_protocol_changed_mixed
+                Conversation.Protocol.MLS -> R.string.label_system_message_conversation_protocol_changed_mls
+            }
+        )
+
+        class HistoryLost: SystemMessage(
+            R.drawable.ic_info,
+            R.string.label_system_message_conversation_history_lost,
+            true)
+
+        class HistoryLostProtocolChanged: SystemMessage(
```

Comment:
`HistoryLostProtocolChanged` and `HistoryLost` could be just `object`  not `class`

Reply:
Thanks for the review! I'm mainly trying to produce a build for Antje here, I'll open real PR later when this is ready to be merged.